### DISCUSSION TITLE
Accessibility concerns with Block Content modals

### DISCUSSION THREAD

**Discussion by tinykite** [sanity-io/sanity#3439] [Sanity Studio Feedback]
When triggering a modal to open the image upload/editing interface within the block content editor (using the "edit" button that appears in the below screenshot), I noticed there is no focus management. A modal will open, but it will not trap a user's focus within the modal.

![Screen Shot 2022-07-24 at 5 11 16 PM](https://user-images.githubusercontent.com/4039311/180667807-93bdc601-97c9-48ef-8b4d-898c393366a4.png)

To be able to access the buttons within the modals (after toggling one with my keyboard) I needed to tab many times — through partially-obscured content behind the modal — to be able to access the buttons within the modal.

A common practice for accessibility is to capture focus within the modal dialog once it appears (so that a user won't accidentally navigate out of a modal, or have difficulty navigating into it). 

Generally speaking, this also goes hand-in-hand with locking scrolling of the page, and focusing the first logically focusable element in the modal. Once the modal closes, focus should return to the button that originally triggered the modal. 

There is a general example of this pattern in [Adrian Roselli's blog ](https://adrianroselli.com/2020/10/dialog-focus-in-screen-readers.html), and it is a pattern that has also been adopted by the [Adobe react-aria library](https://react-spectrum.adobe.com/react-aria/useModal.html) and the [Reach UI library](https://reach.tech/dialog/), among other popular options for implementing accessible modals.

**Comment by KittyGiraudel** [comment]
I can also recommend [a11y-dialog](https://a11y-dialog.netlify.app/), an accessible dialog library I have been maintaining since 2016 which is [recommended by Scott O'hara](https://www.scottohara.me/blog/2019/03/05/open-dialog.html) and Adrian Roselli. It comes with [a React port](github.com/kittyGiraudel/react-a11y-dialog/) which I also maintain. I have also authored a pretty [comprehensive breakdown of react-a11y-dialog vs. other alternatives](https://github.com/KittyGiraudel/react-a11y-dialog/issues/58) (such as `@reach/dialog`, `@react-aria/dialog` and `@headlessui/react`).
